sandman/source/logger.cpp

```diff
@@ -1,186 +1,65 @@
 #include "logger.h"
 
-#if defined (__linux__)
-	#include "ncurses_ui.h"
-#endif // defined (__linux__)
+std::mutex Logger::ms_Mutex;
+std::ofstream Logger::ms_File;
+Logger Logger::ms_Logger(ms_File);
 
-#include <cstdio>
-#include <cstring>
-#include <ctime>
-#include <mutex>
-
-// Locals
-//
-
-// Used to enforce serialization of messages.
-std::mutex s_LogMutex;
-
-// The file to log messages to.
-static std::FILE* s_LogFile = nullptr;
-
-// Whether to echo messages to the screen.
-static bool s_LogToScreen = false;
-
-// Functions
-//
-
-// Initialize the logger.
-//
-// p_LogFileName:	File name of the log for output.
-//
-// returns:		True if successful, false otherwise.
-//
-bool LoggerInitialize(char const* p_LogFileName)

```

Comment:
The documentation comment for `::Logger::Initialize` is in `logger.h`.
```C++
	/// @brief Initializes the global logger such that it can write
	/// to the file denoted by the passed in file name. If the
	/// file doesn't exist, then it is automatically created.
	///
	/// @warning This does not initialize the shell graphics system.
	///
	/// @returns `true` on success, `false` otherwise.
	[[nodiscard]] static bool Initialize(char const* const logFileName);
```